controller/src/beerocks/master/db/db.h

```diff
@@ -589,6 +589,158 @@ class db {
+     */
+    bool set_client_initial_radio(const sMacAddr &mac, const sMacAddr &initial_radio_mac,
+                                  bool set_persistent_db = true);
+
+    /**
+     * @brief Get the client's initial-radio.
+     * 
+     * @param mac MAC address of client.
+     * 
+     * @return mac of radio that the client has initially connected to.
+     */
+    sMacAddr get_client_initial_radio(const sMacAddr &mac);
+
+    /**
+     * @brief Set the client's stay-on-selected-band.
+     * 
+     * @param mac MAC address of client.
+     * @param stay_on_selected_band: enable client stay on the selected band/bands.
+     * @param set_persistent_db: set-persistent-db flag - default is true (write-through).
+     * 
+     * @return true on success.
+     * @return false on failure.
+     */
+    bool set_client_stay_on_selected_band(const sMacAddr &mac, bool stay_on_selected_band,
+                                          bool set_persistent_db = true);
+
+    /**
+     * @brief Get the client's stay-on-selected-band.
+     * 
+     * @param mac MAC address of client.
+     * 
+     * @return enable client stay on the selected band/bands.
+     */
+    beerocks::ePersistentParamBool get_client_stay_on_selected_band(const sMacAddr &mac);
+
+    /**
+     * @brief Set the client's selected-bands.
+     * 
+     * @param mac MAC address of client.
+     * @param selected_bands: client selected band/bands. FREQ_UNKNOWN is considered as "not-configured"
+     * @param set_persistent_db: set-persistent-db flag - default is true (write-through).
+     * 
+     * @return true on success.
+     * @return false on failure.
+     */
+    bool set_client_selected_bands(const sMacAddr &mac, beerocks::eFreqType selected_bands,
+                                   bool set_persistent_db = true);
+
+    /**
+     * @brief Get the client's selected-bands.
+     * 
+     * @param mac MAC address of client.
+     * 
+     * @return selected band/bands.
+     */
+    beerocks::eFreqType get_client_selected_bands(const sMacAddr &mac);
+
+    /**
+     * @brief Clear client's persistent information.
+     * 
+     * @param mac MAC address of client.
+     * 
+     * @return true on success.
+     * @return false on failure.
+     */
+    bool clear_client_persistent_db(const sMacAddr &mac);
+
+    /**
+     * @brief Update client's persistent information with the runtime information.
+     * 
+     * @param mac MAC address of client.
+     * 
+     * @return true on success.
+     * @return false on failure.
+     */
+    bool update_client_persistent_db(const sMacAddr &mac);
+
+    /**
+     * @brief Get all clients from persistent db
+     * 
+     * @return unordered map of clients, for each client unordered map of params as key-value
+     * @return an empty map of clients is returned if the persistent db is empty
+     */
+    std::unordered_map<std::string, std::unordered_map<std::string, std::string>>
+    get_all_persistent_db_clients();
```

Comment:
Please add dots at the end of the sentences.
Second, maybe you could define a type for this return value with Doxygen documentation and then use it here. This could make the return value explanation much more simple.
Also, when using MAC address please use sMacAddr and not std::string.

Reply:
done